STYLE.md

```diff
@@ -0,0 +1,4068 @@
+</tbody></table>
+
+#### Specifying Slice Capacity
+
+Where possible, provide capacity hints when initializing slices with `make()`,
+particularly when appending.
+
+```go
+make([]T, length, capacity)
+```
+
+Unlike maps, slice capacity is not a hint: the compiler will allocate enough
+memory for the capacity of the slice as provided to `make()`, which means that
+subsequent `append()` operations will incur zero allocations (until the length
+of the slice matches the capacity, after which any appends will require a resize
+to hold additional elements).
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+for n := 0; n < b.N; n++ {
+  data := make([]int, 0)
+  for k := 0; k < size; k++{
+    data = append(data, k)
+  }
+}
+```
+
+</td><td>
+
+```go
+for n := 0; n < b.N; n++ {
+  data := make([]int, 0, size)
+  for k := 0; k < size; k++{
+    data = append(data, k)
+  }
+}
+```
+
+</td></tr>
+<tr><td>
+
+```plain
+BenchmarkBad-4    100000000    2.48s
+```
+
+</td><td>
+
+```plain
+BenchmarkGood-4   100000000    0.21s
+```
+
+</td></tr>
+</tbody></table>
+
+## Style
+
+### Avoid overly long lines
+
+Avoid lines of code that require readers to scroll horizontally
+or turn their heads too much.
+
+We recommend a soft line length limit of **99 characters**.
+Authors should aim to wrap lines before hitting this limit,
+but it is not a hard limit.
+Code is allowed to exceed this limit.
+
+### Be Consistent
+
+Some of the guidelines outlined in this document can be evaluated objectively;
+others are situational, contextual, or subjective.
+
+Above all else, **be consistent**.
+
+Consistent code is easier to maintain, is easier to rationalize, requires less
+cognitive overhead, and is easier to migrate or update as new conventions emerge
+or classes of bugs are fixed.
+
+Conversely, having multiple disparate or conflicting styles within a single
+codebase causes maintenance overhead, uncertainty, and cognitive dissonance,
+all of which can directly contribute to lower velocity, painful code reviews,
+and bugs.
+
+When applying these guidelines to a codebase, it is recommended that changes
+are made at a package (or larger) level: application at a sub-package level
+violates the above concern by introducing multiple styles into the same code.
+
+### Group Similar Declarations
+
+Go supports grouping similar declarations.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+import "a"
+import "b"
+```
+
+</td><td>
+
+```go
+import (
+  "a"
+  "b"
+)
+```
+
+</td></tr>
+</tbody></table>
+
+This also applies to constants, variables, and type declarations.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+
+const a = 1
+const b = 2
+
+
+
+var a = 1
+var b = 2
+
+
+
+type Area float64
+type Volume float64
+```
+
+</td><td>
+
+```go
+const (
+  a = 1
+  b = 2
+)
+
+var (
+  a = 1
+  b = 2
+)
+
+type (
+  Area float64
+  Volume float64
+)
+```
```

Comment:
This is actually enforced by `make lint`

Reply:
Should remove?